File: server/src/main/java/com/fluffy/exam/domain/QuestionType.java
package com.fluffy.exam.domain;

import com.fluffy.global.exception.NotFoundException;
import java.util.Arrays;
import lombok.Getter;

@Getter
public enum QuestionType {

    SHORT_ANSWER,
    LONG_ANSWER,
    SINGLE_CHOICE,
    MULTIPLE_CHOICE,
    TRUE_OR_FALSE;

    public static QuestionType from(String value) {
        return Arrays.stream(values())
                .filter(it -> it.name().equalsIgnoreCase(value))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("존재하지 않는 질문 유형입니다. 질문 유형: " + value));
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/QuestionType.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.NotFoundException

enum class QuestionType {

    SHORT_ANSWER,
    LONG_ANSWER,
    SINGLE_CHOICE,
    MULTIPLE_CHOICE,
    TRUE_OR_FALSE,
    ;

    companion object {
        fun from(question: String): QuestionType {
            return entries.find { it.name.equals(question, ignoreCase = true) }
                ?: throw NotFoundException("존재하지 않는 문제 유형입니다. 문제 유형: $question")
        }
    }
}